engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -519,29 +519,71 @@ private boolean isNothing(Node lib) {
   private Map<String, Function> methods() {
     if (methods == null) {
       CompilerDirectives.transferToInterpreter();
-      var allMethods = new HashMap<String, Function>();
-      var defScope = definitionScope.asModuleScope();
-      var methodsFromThisScope = defScope.getMethodsForType(this);
-      if (methodsFromThisScope != null) {
-        methodsFromThisScope.forEach(
+      methods = getMethods(true);
+    }
+    return methods;
+  }
+
+  /**
+   * Returns methods (both instance and static) defined on this type, including the ones inherited
+   * from Any. Instance methods are defined on this type, static methods are defined on its {@link
```

Comment:
Does this work properly for `Any`, `Number`, `Integer` or `Float` hierarchy? E.g. are methods from `Number` (if any) also included?

Reply:
Good point. It does not - tests added in 9b98189ed11de34ad92ae391839f2327aa840d42 fail. More specifically, the test fails at https://github.com/enso-org/enso/blob/9b98189ed11de34ad92ae391839f2327aa840d42/engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/meta/MetaTypeMethodsTest.java#L103, where methods from `Number` are checked.